src/selector-parse.ts

```diff
@@ -25,4 +28,21 @@ export function getAttributeAndValueInSelector(selector: string) {
 export function attrValueContainsQuotes(value: string) {
     return (value[0] === SelectorSyntax.DOUBLE_QUOTE || value[0] === SelectorSyntax.SINGLE_QUOTE)
 					&& (value.slice(-1) === SelectorSyntax.DOUBLE_QUOTE || value.slice(-1) === SelectorSyntax.SINGLE_QUOTE);
+}
+
+export function parseBlockQuerySelector(selector: string): BlockQueryProperties {
+    let value = selector;
+    let isType = false;
+    
+
+    if (selector.includes(SelectorSyntax.TYPE_SELECTOR)) {
+        isType = true
+        value = selector.substring(1);
+    };
+
+    return {
+        attr: 'opcode',
+        value,
+        isType
+    }
 }
```

Comment:
Small thing, but would be good to have a new line here. Depending on your code editor, there are usually plugins to add a new line at the end of a file on save, so you don't have to think about it. Maybe this will be fix for when you're implementing the linter as well.